akka-actor/src/main/java/akka/dispatch/AbstractBoundedNodeQueue.java

```diff
@@ -167,9 +167,17 @@ public final T poll() {
             final Node<T> deq = getDeq();
             final Node<T> next = deq.next();
             if (next != null) {
+                //Can't rely on next.value after casDeq because 'next' can be popped by another thread already
+                //before this one finishes. 
```

Comment:
Ah, I needed to write this down plus an external interruption to understand you properly, forget my objection, continuing to think.